# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
# Commit: 5288094fbef3
# Full Hash: 5288094fbef3f1d0e895c65bebc2f2b33d7b2bfb
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Fixes to MediaPipelineFilter. r=ng
#   
#   Upstream's implementation for rid and mid assumes they are always present but
#   the current code in MediaPipelineFilter had them optional. We need to double
#   check which way it should be.
# ==============================================================================

diff -r 0aa9398df7c0 -r 5288094fbef3 media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
--- a/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Mon Nov 09 15:03:01 2020 -0500
+++ b/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Mon Nov 09 15:26:46 2020 -0500
@@ -523,9 +523,7 @@
     webrtc::RTPHeader header;
     header.ssrc = ssrc;
     header.payloadType = payload_type;
-    mid.apply([&](const auto& mid) {
-      header.extension.mid.Set(mid.c_str(), mid.length());
-    });
+    mid.apply([&](const auto& mid) { header.extension.mid = mid; });
     return filter.Filter(header);
   }
 };
